src/dlr_relayvm.cc

```diff
@@ -48,10 +59,42 @@ void RelayVMModel::SetupVMModule() {
                     "to override TVM allocations. Using default allocators.";
   }
 
-  tvm::runtime::Module lib = tvm::runtime::Module::LoadFromFile(path_->model_lib);
-  std::ifstream relay_ob(path_->relay_executable, std::ios::binary);
-  std::string code_data((std::istreambuf_iterator<char>(relay_ob)),
-                        std::istreambuf_iterator<char>());
+  std::string code_data;
+  std::string model_lib_path;
+  std::string metadata_path;
+  for (DLRModelElem el : model_elems) {
+    if (el.type == DLRModelElemType::RELAY_EXEC) {
+      if (el.path != nullptr) {
+        std::ifstream relay_ob(el.path, std::ios::binary);
+        code_data.assign((std::istreambuf_iterator<char>(relay_ob)),
+                         std::istreambuf_iterator<char>());
+      } else if (el.data != nullptr && el.data_size > 0) {
+        code_data.assign(static_cast<const char*>(el.data), el.data_size);
+      } else {
+        throw dmlc::Error("Invalid RelayVM model element RELAY_EXEC");
+      }
+    } else if (el.type == DLRModelElemType::TVM_LIB) {
+      if (el.path != nullptr) {
+        model_lib_path = el.path;
+      } else {
+        throw dmlc::Error("Invalid RelayVM model element TVM_LIB");
+      }
+    } else if (el.type == DLRModelElemType::NEO_METADATA) {
+      if (el.path != nullptr) {
+        metadata_path = el.path;
+      } else {
```

Comment:
metadata can be passed directly as a string too, not just as a path

Reply:
Added that.